backend/routes/publicRoutes.js

```diff
@@ -1,21 +1,46 @@
 const publicRouter = require('express').Router();
 const validateWithRules = require('../services/validation');
-const User = require('../models/User');
+const PasswordReset = require('../models/PasswordReset');
+const Crypto = require('crypto');
 
 publicRouter.get("/test", (req, res) => {
    res.send("API is running...");
 });
 
-publicRouter.post("/forgot-password", (req, res) => {
+publicRouter.post("/forgot-password", async (req, res) => {
  // generate one time key
- // store key in database with expiration
+ let key = Crypto.randomBytes(20).toString('hex');
+ 
+ // store key in database with expiration (or update existing based on email)
+ let reset = new PasswordReset({
+      email: req.body.email,
+      key: key,
+      expires: new Date(Date.now() + (3600 * 1000 * 24)),
+  });
+ try {
+   await reset.save();
+   return res.status(201).json({ message: "http://localhost:3000/reset-password/" + key }); // TODO: Don't leave me here
```

Comment:
We could somehow determine the proper URL dynamically to avoid hardcoding it, at the very least the port could be injected via process.env.port

Reply:
Also I think the URL might be missing a /api/ before reset-password

Reply:
it's supposed to be to the frontend as it is- this is the link that goes in the email